### PR TITLE
sql: fix the pretty-printing of CREATE EXTENSION

### LINKED ISSUE #101978: sql: CREATE EXTENSION does not pretty-print properly
**Describe the problem**

CREATE EXTENSION accepts identifiers, which may contain punctuation. However, this is not pretty-printed properly.

**To Reproduce**

`CREATE EXTENSION "a-b"`

Jira issue: CRDB-27197
Epic: CRDB-28893

### PR DESCRIPTION
**Pull request by knz**
Fixes #101978.
Epic: CRDB-28893

Release note: None
Epic: None

### CODE REVIEW

File: pkg/sql/sem/tree/create.go [hunk 1]
```diff
@@ -2147,7 +2147,7 @@ func (node *CreateExtension) Format(ctx *FmtCtx) {
 	// do not contain sensitive information and
 	// 2) we want to get telemetry on which extensions
 	// users attempt to load.
-	ctx.WriteString(node.Name)
```
Comment (otan) [inline pkg/sql/sem/tree/create.go:2150]: `ctx.FormatNode(node.Name)` is better right?

**Review by otan — APPROVED** [review]
optional change

File: pkg/sql/sem/tree/create.go (hunk 1 above)
Comment (knz) [reply pkg/sql/sem/tree/create.go:2150]: `FormatNode` removes the identifier when performing redaction. Which is exactly what we don't want here.

**Comment by knz** [comment]
bors r=otan

**Pull request merged**